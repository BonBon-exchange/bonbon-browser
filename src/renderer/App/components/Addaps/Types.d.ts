export interface AddapsProps {
  boardId?: string;
}
