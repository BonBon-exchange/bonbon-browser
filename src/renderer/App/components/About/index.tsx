import { About } from './About';

export * from './About';
export default About;
