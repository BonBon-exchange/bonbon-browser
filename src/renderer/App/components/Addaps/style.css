#Addaps__background {
  background: linear-gradient(
    200.96deg,
    #fedc2a -29.09%,
    #dd5789 51.77%,
    #7a2c9e 129.35%
  );
  height: 100%;
  width: 100%;
  position: fixed;
}

#Minimap__detection-zone {
  height: 100%;
  width: 20px;
  background: transparent;
  position: fixed;
  right: 0;
}
