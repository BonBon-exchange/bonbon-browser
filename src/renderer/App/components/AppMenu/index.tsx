import { AppMenu } from './AppMenu';

export * from './AppMenu';
export default AppMenu;
