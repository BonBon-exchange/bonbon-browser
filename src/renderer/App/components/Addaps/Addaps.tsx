/* eslint-disable promise/always-return */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/prefer-default-export */
/* eslint-disable no-use-before-define */
import React, { useCallback, useEffect, useState, lazy, Suspense } from 'react';
import clsx from 'clsx';
import { useTranslation } from 'react-i18next';

import { useGlobalEvents } from 'renderer/App/hooks/useGlobalEvents';
import { useStoreHelpers } from 'renderer/App/hooks/useStoreHelpers';
import { useAppSelector } from 'renderer/App/store/hooks';
import { Loader } from 'renderer/App/components/Loader';
import { About } from 'renderer/App/components/About';

import { AddapsProps } from './Types';

import './style.css';

const Board = lazy(() => import('renderer/App/components/Board'));
const LeftBar = lazy(() => import('renderer/App/components/LeftBar'));
const DownloadsPreview = lazy(
  () => import('renderer/App/components/DownloadsPreview')
);
const Settings = lazy(() => import('renderer/App/components/Settings'));
const Bookmarks = lazy(() => import('renderer/App/components/Bookmarks'));
const History = lazy(() => import('renderer/App/components/History'));
const Downloads = lazy(() => import('renderer/App/components/Downloads'));
const Extensions = lazy(() => import('renderer/App/components/Extensions'));
const Documentation = lazy(
  () => import('renderer/App/components/Documentation')
);
const Popup = lazy(() => import('renderer/App/components/Popup'));
const AppMenu = lazy(() => import('renderer/App/components/AppMenu'));
const Minimap = lazy(() => import('renderer/App/components/Minimap'));

export const Addaps: React.FC<AddapsProps> = ({ boardId }) => {
  useGlobalEvents();
  const { items } = useAppSelector((state) => state.downloads);
  const { board } = useStoreHelpers({ boardId });
  const [showPopup, setShowPopup] = useState<boolean>(false);
  const [showSettings, setShowSettings] = useState<boolean>(false);
  const [showBookmarks, setShowBookmarks] = useState<boolean>(false);
  const [showHistory, setShowHistory] = useState<boolean>(false);
  const [showDownloads, setShowDownloads] = useState<boolean>(false);
  const [showDocumentation, setShowDocumentation] = useState<boolean>(false);
  const [showExtensions, setShowExtensions] = useState<boolean>(false);
  const [showAppMenu, setShowAppMenu] = useState<boolean>(false);
  const [showDownloadsPreview, setShowDownloadsPreview] =
    useState<boolean>(false);
  const [popupTitle, setPopupTitle] = useState<string>('');
  const [popupChildren, setPopupChildren] = useState<JSX.Element>();
  const [showMinimap, setShowMinimap] = useState<boolean>(false);
  const { i18n } = useTranslation();

  const showAppMenuAction = useCallback(() => {
    setShowAppMenu(!showAppMenu);
    setShowDownloadsPreview(false);
  }, [showAppMenu]);

  const showDownloadsPreviewAction = useCallback(() => {
    if (items.length > 0) {
      setShowDownloadsPreview(!showDownloadsPreview);
      setShowAppMenu(false);
    } else {
      // show downloads
      setShowDownloads(true);
    }
  }, [showDownloadsPreview, items.length]);

  const windowClickListener = () => {
    setShowAppMenu(false);
    setShowDownloadsPreview(false);
    window.app.tools.clicked();
  };

  const minimapMouseEnterListener = () => {
    setShowMinimap(true);
  };

  const showAbout = () => {
    setPopupChildren(<About />);
    setPopupTitle('About');
    setShowPopup(!showPopup);
    window.app.analytics.event('open_about');
  };

  const handleCloseSettings = () => setShowSettings(false);
  const handleCloseBookmarks = () => setShowBookmarks(false);
  const handleCloseHistory = () => setShowHistory(false);
  const handleCloseDownloads = () => setShowDownloads(false);
  const handleCloseDocumentation = () => setShowDocumentation(false);
  const handleCloseExtensions = () => setShowExtensions(false);

  const handleShowSettings = () => setShowSettings(true);
  const handleShowBookmarks = () => setShowBookmarks(true);
  const handleShowHistory = () => setShowHistory(true);
  const handleShowDownloads = () => setShowDownloads(true);
  const handleShowDocumentation = () => setShowDocumentation(true);
  const handleShowExtensions = () => setShowExtensions(true);

  useEffect(() => {
    if (boardId) board.load({ id: boardId });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [boardId]);

  useEffect(() => {
    window.app.listener.showAppMenu(showAppMenuAction);
    return () => window.app.off.showAppMenu();
  }, [showAppMenuAction]);

  useEffect(() => {
    window.app.listener.showDownloadsPreview(showDownloadsPreviewAction);
    return () => window.app.off.showDownloadsPreview();
  }, [showDownloadsPreviewAction]);

  useEffect(() => {
    window.addEventListener('click', windowClickListener);
    return () => window.removeEventListener('click', windowClickListener);
  }, []);

  useEffect(() => {
    document
      .querySelector('#Minimap__detection-zone')
      ?.addEventListener('mouseenter', minimapMouseEnterListener);
    return () =>
      document
        .querySelector('#Minimap__detection-zone')
        ?.removeEventListener('mouseenter', minimapMouseEnterListener);
  }, []);

  return (
    <Suspense fallback={<Loader />}>
      <div
        id="Addaps__container"
        className={clsx({
          'justify-content-right':
            i18n.language === 'ar' || i18n.language === 'fa',
        })}
      >
        <div id="Addaps__background" />
        <LeftBar />
        <Board
          boardId={boardId}
          isFullSize={
            showSettings ||
            showBookmarks ||
            showHistory ||
            showDownloads ||
            showDocumentation ||
            showExtensions
          }
        />
        {showPopup && (
          <Popup title={popupTitle} closePopup={() => setShowPopup(false)}>
            {popupChildren}
          </Popup>
        )}
        {showAppMenu && (
          <AppMenu
            showAbout={showAbout}
            showSettings={handleShowSettings}
            showBookmarks={handleShowBookmarks}
            showHistory={handleShowHistory}
            showDownloads={handleShowDownloads}
            showDocumentation={handleShowDocumentation}
            showExtensions={handleShowExtensions}
          />
        )}
        {showDownloadsPreview && <DownloadsPreview />}
        {showSettings && <Settings handleClose={handleCloseSettings} />}
        {showBookmarks && <Bookmarks handleClose={handleCloseBookmarks} />}
        {showHistory && <History handleClose={handleCloseHistory} />}
        {showDownloads && <Downloads handleClose={handleCloseDownloads} />}
        {showExtensions && <Extensions handleClose={handleCloseExtensions} />}
        {showDocumentation && (
          <Documentation handleClose={handleCloseDocumentation} />
        )}
        <div id="Minimap__detection-zone" />
        {showMinimap && <Minimap handleHide={() => setShowMinimap(false)} />}
      </div>
    </Suspense>
  );
};
