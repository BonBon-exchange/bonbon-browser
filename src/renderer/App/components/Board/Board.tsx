/* eslint-disable promise/always-return */
/* eslint-disable @typescript-eslint/ban-ts-comment */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/prefer-default-export */
/* eslint-disable no-use-before-define */
import clsx from 'clsx';
import { AnimatePresence } from 'framer-motion';
import React, { useCallback, useEffect, useState } from 'react';

import { Browser } from 'renderer/App/components/Browser';
import { BrowserProps } from 'renderer/App/components/Browser/Types';
import { useBoard } from 'renderer/App/hooks/useBoard';
import { useBrowserMethods } from 'renderer/App/hooks/useBrowserMethods';
import { useStoreHelpers } from 'renderer/App/hooks/useStoreHelpers';
import { useAppDispatch } from 'renderer/App/store/hooks';
import {
  setActiveBrowser,
  setBoardHeight,
} from 'renderer/App/store/reducers/Board';

import { BoardProps } from './Types';

import './style.scss';

export const Board: React.FC<BoardProps> = ({ isFullSize, boardId }) => {
  const board = useBoard();
  const dispatch = useAppDispatch();
  const { focus } = useBrowserMethods();
  const [items, setItems] = useState<BrowserProps[]>([]);
  const helpers = useStoreHelpers();

  const boardContainer = document.querySelector('#Board__container');

  const makeBrowsers = useCallback((sorted: BrowserProps[]) => {
    return sorted.map((b) => <Browser {...b} key={b.id} firstRendering />);
  }, []);

  const contextMenuListener = (e: MouseEvent) => {
    e.preventDefault();
    window.app.tools.showBoardContextMenu({ x: e.clientX, y: e.clientY });
  };

  useEffect(() => {
    if (!board.isFullSize) {
      setTimeout(() => {
        helpers.board
          .distributeWindowsByOrder(board.browsers)
          .then(() => {
            setTimeout(
              () => board.activeBrowser && focus(board.activeBrowser),
              300
            );
          })
          .catch(console.log);
      }, 300);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [board.isFullSize]);

  useEffect(() => {
    const toSort = [...board.browsers];
    const sorted = toSort.sort((a, b) => {
      return a.id > b.id ? 1 : -1;
    });
    setItems(sorted);

    if (boardContainer) {
      const max = board.browsers.reduce(
        (acc, val) => Math.max(acc, val.top + val.height + 100),
        0
      );

      // @ts-ignore
      boardContainer.style.height = `${Number(max)}px`;

      dispatch(setBoardHeight(max));
    }
  }, [board.browsers, dispatch, boardContainer]);

  useEffect(() => {
    window.app.board.setWindowsCount({
      boardId: board.id,
      count: board.browsers.length,
    });
  }, [board.browsers.length, board.id]);

  useEffect(() => {
    if (board.activeBrowser) {
      focus(board.activeBrowser, true);
      dispatch(setActiveBrowser(board.activeBrowser));
    }
  }, [board.activeBrowser, focus, dispatch]);

  // focus activeBrowser when componentDidMount
  useEffect(() => {
    setTimeout(() => {
      if (board.activeBrowser) focus(board.activeBrowser);
      if (boardId) {
        window.app.browser
          .getUrlToOpen()
          .then((res) => {
            if (res) helpers.browser.add({ url: res });
          })
          .catch(console.log);
      }
    }, 0);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    document
      .getElementById('Board__container')
      ?.addEventListener('contextmenu', contextMenuListener);

    return () =>
      document
        .getElementById('Board__container')
        ?.removeEventListener('contextmenu', contextMenuListener);
  }, []);

  return (
    <div
      id="Board__container"
      className={clsx({
        'Board__is-maximized': board.isFullSize || isFullSize,
        'Board__isnt-maximized': !board.isFullSize && !isFullSize,
      })}
    >
      <AnimatePresence>{makeBrowsers(items)}</AnimatePresence>
      <div className="Board__edge-snap-left" />
      <div className="Board__edge-snap-right" />
      <div className="Board__edge-snap-maximized" />
    </div>
  );
};
