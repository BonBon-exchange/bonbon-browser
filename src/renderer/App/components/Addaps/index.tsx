import { Addaps } from './Addaps';

export * from './Addaps';
export default Addaps;
