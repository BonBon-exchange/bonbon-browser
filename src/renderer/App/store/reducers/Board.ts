/* eslint-disable @typescript-eslint/no-unused-expressions */
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { v4 } from 'uuid';

import { WritableDraft } from 'immer/dist/internal';
import { BoardType } from 'renderer/App/components/Board/Types';
import { BrowserProps } from 'renderer/App/components/Browser/Types';

type UpdateBrowserType = {
  browserId: string;
  params: Partial<BrowserProps>;
};

type UpdateBrowserUrlType = {
  url: string;
  browserId: string;
};

type UpdateBrowserLoadingType = {
  isLoading: boolean;
  browserId: string;
};

type UpdateBrowserTitleType = {
  title: string;
  browserId: string;
};

type UpdateBrowserFavType = {
  favicon: string;
  browserId: string;
};

type UpdateBrowserCertificateFingerprintType = {
  certificateErrorFingerprint: string | null;
  browserId: string;
};

interface BoardState {
  board: BoardType;
}

const boardId = v4();
const newBoard = {
  id: boardId,
  label: 'New collection',
  browsers: [],
  activeBrowser: null,
  closedUrls: [],
  isFullSize: false,
  browsersActivity: [],
  height: 0,
};

export const initialState: BoardState = {
  board: newBoard,
};

const addCLosedUrl = (state: WritableDraft<BoardState>, url: string) => {
  if (state.board.closedUrls.length > 20) {
    state.board.closedUrls.splice(0, 1);
  }
  state.board.closedUrls.push(url);
};

const addBrowserActivity = (
  state: WritableDraft<BoardState>,
  browserId: string
) => {
  if (state.board.browsersActivity.length > 20) {
    state.board.browsersActivity.splice(0, 1);
  }
  state.board.browsersActivity = state.board.browsersActivity.filter(
    (b) => b !== browserId
  );
  state.board.browsersActivity.push(browserId);
};

const findAndSelectTabForExtensionsFromBrowserId = (
  state: WritableDraft<BoardState>,
  browserId: string
): void => {
  const wcId = state.board.browsers.find(
    (b) => b.id === browserId
  )?.webContentsId;
  if (wcId) window.app.browser.select(wcId);
};

const setStateActiveBrowser = (
  state: WritableDraft<BoardState>,
  browserId: string
): void => {
  state.board.activeBrowser = browserId;
  addBrowserActivity(state, browserId);
  findAndSelectTabForExtensionsFromBrowserId(state, browserId);
};

const getBrowserIndexFromBrowserId = (
  state: WritableDraft<BoardState>,
  browserId: string
): number => {
  return state.board.browsers.findIndex((b) => b.id === browserId);
};

export const boardSlice = createSlice({
  name: 'board',
  initialState,
  reducers: {
    setBoard: (state, action: PayloadAction<BoardType>) => {
      state.board = action.payload;
      const activeBrowser = state.board.browsers.find(
        (b) => b.id === state.board.activeBrowser
      );
      if (!state.board.activeBrowser || !activeBrowser) {
        if (state.board.browsers.length > 0) {
          setStateActiveBrowser(
            state,
            state.board.browsers[state.board.browsers.length - 1].id
          );
        }
      }
      if (state.board.activeBrowser) {
        findAndSelectTabForExtensionsFromBrowserId(
          state,
          state.board.activeBrowser
        );
      }
    },
    toggleBoardFullSize: (state) => {
      state.board.isFullSize = !state.board.isFullSize;
      window.app.analytics.event(`toggle_fullsize_${!state.board.isFullSize}`);
    },
    setActiveBrowser: (state, action: PayloadAction<string>) => {
      setStateActiveBrowser(state, action.payload);
    },
    addBrowser: (state, action: PayloadAction<BrowserProps>) => {
      if (action.payload.browserIndex) {
        state.board.browsers.splice(
          action.payload.browserIndex,
          0,
          action.payload
        );
      } else {
        state.board.browsers.push(action.payload);
      }
      setStateActiveBrowser(state, action.payload.id);
      window.app.analytics.event('add_browser');
    },
    setBrowsers: (state, action: PayloadAction<BrowserProps[]>) => {
      state.board.browsers = action.payload;
    },
    updateBrowser: (state, action: PayloadAction<UpdateBrowserType>) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex] = {
          ...state.board.browsers[browserIndex],
          ...action.payload.params,
        };
      }
    },
    updateBrowserUrl: (state, action: PayloadAction<UpdateBrowserUrlType>) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].url = action.payload.url;
      }
    },
    updateBrowserLoading: (
      state,
      action: PayloadAction<UpdateBrowserLoadingType>
    ) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].isLoading = action.payload.isLoading;
      }
    },
    updateBrowserTitle: (
      state,
      action: PayloadAction<UpdateBrowserTitleType>
    ) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].title = action.payload.title;
      }
    },
    updateBrowserFav: (state, action: PayloadAction<UpdateBrowserFavType>) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].favicon = action.payload.favicon;
      }
    },
    renameBoard: (state, action: PayloadAction<string>) => {
      state.board.label = action.payload;
      window.app.analytics.event('rename_board');
    },
    minimizeBrowser: (state, action: PayloadAction<string>) => {
      const browserIndex = getBrowserIndexFromBrowserId(state, action.payload);
      state.board.browsers[browserIndex].isMinimized = true;
    },
    unminimizeBrowser: (state, action: PayloadAction<string>) => {
      const browserIndex = getBrowserIndexFromBrowserId(state, action.payload);

      state.board.browsers[browserIndex].isMinimized = false;
    },
    removeBrowser: (state, action: PayloadAction<string>) => {
      const browserIndex = getBrowserIndexFromBrowserId(state, action.payload);

      // remove browser from state
      if (browserIndex > -1) {
        if (!state.board.closedUrls) state.board.closedUrls = [];
        addCLosedUrl(state, state.board.browsers[browserIndex].url);
        state.board.lastClosedBrowserDimensions = [
          state.board.browsers[browserIndex].width,
          state.board.browsers[browserIndex].height,
        ];
        state.board.browsers.splice(browserIndex, 1);

        // clean browsersActivity state
        state.board.browsersActivity = state.board.browsersActivity.filter(
          (b) => b !== action.payload
        );
      }

      // clean activeBrowser
      if (state.board.activeBrowser === action.payload) {
        if (state.board.browsers.length > 0) {
          setStateActiveBrowser(
            state,
            state.board.browsersActivity[
              state.board.browsersActivity.length - 1
            ]
          );
        } else {
          state.board.activeBrowser = null;
          window.app.browser.selectBrowserView();
        }
      }

      window.app.analytics.event('close_browser');
    },
    removeAllBrowsersExcept: (state, action: PayloadAction<string>) => {
      state.board.browsers
        .filter((b) => b.id !== action.payload)
        .forEach((b) => addCLosedUrl(state, b.url));

      state.board.browsers = state.board.browsers.filter(
        (b) => b.id === action.payload
      );

      // clean activeBrowser
      setStateActiveBrowser(state, action.payload);

      // clean browsersActivity state
      state.board.browsersActivity = state.board.browsersActivity.filter(
        (b) => b === action.payload
      );

      // send event
      window.app.analytics.event('close_others_browser');
    },
    updateBrowserCertificateErrorFingerprint: (
      state,
      action: PayloadAction<UpdateBrowserCertificateFingerprintType>
    ) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].certificateErrorFingerprint =
          action.payload.certificateErrorFingerprint;
      }
    },
    removeAllBrowsers: (state) => {
      if (!state.board.closedUrls) state.board.closedUrls = [];
      state.board.browsers.forEach((b) => addCLosedUrl(state, b.url));
      state.board.browsers = [];
      state.board.browsersActivity = [];
      state.board.activeBrowser = null;
      window.app.browser.selectBrowserView();
      window.app.analytics.event('close_all_browsers');
    },
    removeLastClosedUrl: (state) => {
      state.board.closedUrls.splice(state.board.closedUrls.length - 1, 1);
    },
    toggleSearch: (state, action: PayloadAction<string>) => {
      const browserIndex = getBrowserIndexFromBrowserId(state, action.payload);
      state.board.browsers[browserIndex].isSearching =
        !state.board.browsers[browserIndex].isSearching;
    },
    setLastResizedBrowserDimensions: (
      state,
      action: PayloadAction<[number, number]>
    ) => {
      state.board.lastResizedBrowserDimensions = action.payload;
    },
    setBoardHeight: (state, action: PayloadAction<number>) => {
      state.board.height = action.payload;
    },
  },
});

export const {
  setBoard,
  updateBrowserUrl,
  updateBrowserFav,
  renameBoard,
  removeBrowser,
  minimizeBrowser,
  unminimizeBrowser,
  removeAllBrowsers,
  addBrowser,
  setActiveBrowser,
  updateBrowserTitle,
  updateBrowser,
  toggleBoardFullSize,
  removeLastClosedUrl,
  setBrowsers,
  removeAllBrowsersExcept,
  updateBrowserLoading,
  updateBrowserCertificateErrorFingerprint,
  toggleSearch,
  setLastResizedBrowserDimensions,
  setBoardHeight,
} = boardSlice.actions;

export default boardSlice.reducer;
