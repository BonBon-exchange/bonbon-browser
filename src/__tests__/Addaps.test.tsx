import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Middleware } from '@reduxjs/toolkit';

import { mockWindow } from './beforeAll';
import { Addaps } from '../renderer/App/components/Addaps';
import { initialState } from '../renderer/App/store/reducers/Board';

let store: any;
const middlewares: Middleware[] = [];

describe('Addaps', () => {
  beforeAll(() => {
    mockWindow();
    const mockStore = configureStore(middlewares);
    store = mockStore({ board: initialState });
  });

  it('should render with no boardId', () => {
    expect(
      render(
        <Provider store={store}>
          <Addaps />
        </Provider>
      )
    ).toBeTruthy();
  });

  it('should render with boardId', () => {
    expect(
      render(
        <Provider store={store}>
          <Addaps boardId="someboardid" />
        </Provider>
      )
    ).toBeTruthy();
  });
});
