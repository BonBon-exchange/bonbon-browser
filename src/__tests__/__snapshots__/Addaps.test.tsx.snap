// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addaps should match snapshot with boardId 1`] = `
<div
  className=""
  id="Addaps__container"
>
  <div
    id="Addaps__background"
  />
  <div
    id="LeftBar__browserFavContainer"
  >
    <ul
      style={Object {}}
    >
      <div
        id="LeftBar__browserFavContainerItems"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        id="ButtonAddBrowser"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </ul>
  </div>
  <div
    className="Board__isnt-maximized"
    id="Board__container"
  >
    <div
      className="Board__edge-snap-left"
    />
    <div
      className="Board__edge-snap-right"
    />
    <div
      className="Board__edge-snap-maximized"
    />
  </div>
  <div
    id="Minimap__detection-zone"
  />
</div>
`;

exports[`Addaps should match snapshot with no boardId 1`] = `
<div
  className="Loader__loading"
>
  <img
    src="test-file-stub"
  />
</div>
`;

exports[`Addaps should show and hide App Menu, and match snapshot 1`] = `"<div id=\\"Addaps__container\\" class=\\"\\" style=\\"\\"><div id=\\"Addaps__background\\"></div><div id=\\"LeftBar__browserFavContainer\\"><ul><div id=\\"LeftBar__browserFavContainerItems\\"></div><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"AddIcon\\" id=\\"ButtonAddBrowser\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path></svg></ul></div><div id=\\"Board__container\\" class=\\"Board__isnt-maximized\\" style=\\"height: 0px;\\"><div class=\\"Board__edge-snap-left\\"></div><div class=\\"Board__edge-snap-right\\"></div><div class=\\"Board__edge-snap-maximized\\"></div></div><div id=\\"AppMenu__container\\" class=\\"AppMenu__container\\" data-testid=\\"app-menu\\"><ul><li>Toggle dark mode</li><li>Extensions</li><li>Downloads</li><li>Bookmarks</li><li>History</li><li>Settings</li><li>Documentation</li><li>About</li></ul></div><div id=\\"Minimap__detection-zone\\"></div></div>"`;
